### PR TITLE
[MESOS][SPARK-8798] Allow additional uris to be fetched with mesos

### PR DESCRIPTION
**Pull request by tnachen**
Some users like to download additional files in their sandbox that they can refer to from their spark program, or even later mount these files to another directory.

### CODE REVIEW

**Comment by tnachen** [comment]
@andrewor14 :)

**Comment by tnachen** [comment]
jenkins retest please

**Comment by andrewor14** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala [hunk 1]
```diff
@@ -124,6 +124,13 @@ private[spark] class MesosSchedulerBackend(
         Value.Scalar.newBuilder()
           .setValue(MemoryUtils.calculateTotalMemory(sc)).build())
       .build()
+
+    sc.conf.getOption("spark.mesos.uris").map { uris =>
+      uris.split(",").foreach { uri =>
+        command.addUris(CommandInfo.URI.newBuilder().setValue(uri.trim()))
+      }
+    }
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala]: this is duplicated 3 times. Do you think the refactoring your other patch (roles principals) will help? Should we merge that one first?

Comment (dragos) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala]: There's a common superclass of all these three classes called `MesosSchedulerUtils`. I think this code could go there.

**Comment by andrewor14** [comment]
Merging into master 1.5

**Pull request closed without merging**